token/services/auditor/manager.go

```diff
@@ -110,89 +109,27 @@ func (cm *Manager) restore(tmsID token.TMSID, walletID string) error {
 	if err != nil {
 		return errors.WithMessagef(err, "failed to get network instance for [%s]", tmsID)
 	}
-
 	auditor, err := cm.getAuditor(tmsID, walletID)
 	if err != nil {
 		return errors.WithMessagef(err, "failed to get auditor for [%s]", walletID)
 	}
-	it, err := auditor.db.Transactions(auditdb.QueryTransactionsParams{})
+	it, err := auditor.db.TokenRequests(auditdb.QueryTokenRequestsParams{Statuses: []driver.TxStatus{auditdb.Pending}})
 	if err != nil {
 		return errors.Errorf("failed to get tx iterator for [%s:%s]", tmsID, walletID)
 	}
 	defer it.Close()
-	v, err := net.Vault(tmsID.Namespace)
-	if err != nil {
-		return errors.Errorf("failed to get vault for [%s:%s]", tmsID, walletID)
-	}
-	type ToBeUpdated struct {
-		TxID    string
-		Status  auditdb.TxStatus
-		Message string
-	}
-	var toBeUpdated []ToBeUpdated
-	var pendingTXs []string
 	for {
```

Comment:
A more fluent way of writing this would be
```
var record SomeType
var err error
for record, err = it.Next(); record != nil && err == nil; record, err = it.Next() {
 if err := net.AddFinalityListener(record.TxID, &FinalityListener{net, auditor.db}); err != nil {
  return errors.WithMessagef(err, "failed to subscribe event listener to network [%s] for [%s]", tmsID, record.TokenRequest)
 }
}
if err != nil {
 return errors.Errorf("failed to get next tx record for [%s:%s]", tmsID, walletID)
}
```

Reply:
but the error generated by `record, err = it.Next()` is not checked